packages/legacy/core/App/modules/openid/context/OpenIDCredentialRecordProvider.tsx

```diff
@@ -0,0 +1,151 @@
+import { createContext, PropsWithChildren, useContext, useEffect, useState } from 'react'
+
+import {
+  ClaimFormat,
+  SdJwtVcRecord,
+  SdJwtVcRepository,
+  W3cCredentialRecord,
+  W3cCredentialRepository,
+} from '@credo-ts/core'
+import { useAgent } from '@credo-ts/react-hooks'
+import { recordsAddedByType, recordsRemovedByType } from '@credo-ts/react-hooks/build/recordUtils'
+import { TOKENS, useServices } from '../../../container-api'
+
+type OpenIDCredentialRecord = W3cCredentialRecord | SdJwtVcRecord | undefined
+
+export type OpenIDCredentialContext = {
+  openIdState: OpenIDCredentialRecordState
+  storeCredential: (cred: W3cCredentialRecord | SdJwtVcRecord) => Promise<void>
+  removeCredential: (cred: W3cCredentialRecord | SdJwtVcRecord) => Promise<void>
+}
+
+export type OpenIDCredentialRecordState = {
+  openIDCredentialRecords: Array<OpenIDCredentialRecord>
+  w3cCredentialRecords: Array<W3cCredentialRecord>
+  sdJwtVcRecords: Array<SdJwtVcRecord>
+  isLoading: boolean
+}
+
+const addRecord = (record: W3cCredentialRecord, state: OpenIDCredentialRecordState): OpenIDCredentialRecordState => {
+  const newRecordsState = [...state.w3cCredentialRecords]
+  newRecordsState.unshift(record)
+
+  return {
+    ...state,
+    w3cCredentialRecords: newRecordsState,
+  }
+}
+
+const removeRecord = (record: W3cCredentialRecord, state: OpenIDCredentialRecordState): OpenIDCredentialRecordState => {
+  const newRecordsState = [...state.w3cCredentialRecords]
+  const index = newRecordsState.findIndex((r) => r.id === record.id)
+  if (index > -1) {
+    newRecordsState.splice(index, 1)
+  }
+
+  return {
+    ...state,
+    w3cCredentialRecords: newRecordsState,
+  }
+}
+
+const defaultSate: OpenIDCredentialRecordState = {
+  openIDCredentialRecords: [],
+  w3cCredentialRecords: [],
+  sdJwtVcRecords: [],
+  isLoading: true,
+}
+
+interface Props {
+  children: React.ReactNode
+}
+
+const OpenIDCredentialRecordContext = createContext<OpenIDCredentialContext>(null as unknown as OpenIDCredentialContext)
+
+const isW3CCredentialRecord = (record: W3cCredentialRecord) => {
+  return record.getTags()?.claimFormat === ClaimFormat.JwtVc
+}
+
+const filterW3CCredentialsOnly = (credentials: W3cCredentialRecord[]) => {
+  return credentials.filter((r) => isW3CCredentialRecord(r))
+}
+
+// eslint-disable-next-line react/prop-types
+export const OpenIDCredentialRecordProvider: React.FC<PropsWithChildren<Props>> = ({ children }) => {
+  const [state, setState] = useState<OpenIDCredentialRecordState>(defaultSate)
+
+  const { agent } = useAgent()
+  const [logger] = useServices([TOKENS.UTIL_LOGGER])
+
+  function checkAgent() {
+    if (!agent) {
+      const error = 'Agent undefined!'
+      logger.error(`[OpenIDCredentialRecordProvider][deleteCredential] ${error}`)
```

Comment:
```suggestion
      logger.error(`[OpenIDCredentialRecordProvider] ${error}`)
```
Since this error doesn't seem to be specifically related to the deleteCredential functionality